drivers/eeprom/eeprom_mchp_xec.c

```diff
@@ -0,0 +1,328 @@
+/*
+ * Copyright (c) 2022 Microchip Technology Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT microchip_xec_eeprom
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/eeprom.h>
+#include <soc.h>
+
+#ifdef CONFIG_PINCTRL
+#include <zephyr/drivers/pinctrl.h>
+#endif
```

Comment:
The DTS binding requires `pinctrl-0` and `pinctrl-names` properties. Is the `#ifdef CONFIG_PINCTRL` needed or could the driver just depend on `CONFIG_PINCTRL`?

Reply:
this driver is common between mec172x and mec15xx family of devices. mec15xx doesn't use pinctrl hence we use this ifdef, this is similar to how we have done with all pinctrl based drivers

Reply:
But the `microchip,xec-eeprom` devicetree binding states `pinctrl-0` and `pinctrl-names` as required. If the binding requires it, the driver can too. Otherwise you must change the binding to make those properties optional.

Reply:
will update binding to make properties optional